test/java/seedu/addressbook/commands/DeleteCommandTest.java

```diff
@@ -0,0 +1,169 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.ui.TextUi;
+
+public class DeleteCommandTest {
+
+    private AddressBook emptyAddressBook;
+    private AddressBook addressBook;
+
+    private List<ReadOnlyPerson> emptyDisplayList;
+    private List<ReadOnlyPerson> listWithEveryone;
+    private List<ReadOnlyPerson> listWithSurnameDoe;
+
+    @Before
+    public void setUp() throws Exception {
+        Person johnDoe = new Person(new Name("John Doe"), new Phone("111", false),
+                new Email("john@doe.com", false), new Address("395C Ben Road", false), new UniqueTagList());
+        Person janeDoe = new Person(new Name("Jane Doe"), new Phone("222", false),
+                new Email("jane@doe.com", false), new Address("33G Ohm Road", false), new UniqueTagList());
+        Person davidGrant = new Person(new Name("David Grant"), new Phone("333", false),
+                new Email("david@grant.com", false), new Address("44H Define Road", false),
+                new UniqueTagList());
+
+        emptyAddressBook = new AddressBook();
+
+        addressBook = new AddressBook();
+        addressBook.addPerson(johnDoe);
+        addressBook.addPerson(janeDoe);
+        addressBook.addPerson(davidGrant);
+
+        emptyDisplayList = new ArrayList<ReadOnlyPerson>();
+
+        listWithEveryone = new ArrayList<ReadOnlyPerson>();
+        listWithEveryone.add(johnDoe);
+        listWithEveryone.add(janeDoe);
+        listWithEveryone.add(davidGrant);
+
+        listWithSurnameDoe = new ArrayList<ReadOnlyPerson>();
+        listWithSurnameDoe.add(johnDoe);
+        listWithSurnameDoe.add(janeDoe);
```

Comment:
Consider creating a utility class e.g. `TestUtil` that provides methods to create test objects. e.g. `createAddressBook(Person... persons)`